examples/frontend-links/src/frontend/core.cljs

```diff
@@ -0,0 +1,143 @@
+(ns frontend.core
+  (:require [clojure.string :as string]
+            [fipp.edn :as fedn]
+            [reagent.core :as r]
+            [reitit.coercion :as rc]
+            [reitit.coercion.spec :as rss]
+            [reitit.frontend :as rf]
+            [reitit.frontend.easy :as rfe]
+            [spec-tools.data-spec :as ds]))
+
+;; Components similar to react-router `Link`, `NavLink` and `Redirect`
+;; with Reitit frontend.
+
+(defn home-page []
+  [:div
+   [:h2 "Welcome to frontend"]
+   [:p "This is home page"]])
+
+(defn about-page []
+  [:div
+   [:h2 "About frontend"]
+   [:p "This is about page"]])
+
+(defn redirect!
+  "If `push` is truthy, previous page will be left in history."
+  [{:keys [to path-params query-params push]}]
+  (if push
+    (rfe/push-state to path-params query-params)
+    (rfe/replace-state to path-params query-params)))
+
+(defn Redirect
+  "Component that only causes a redirect side-effect."
+  [props]
+  (redirect! props)
```

Comment:
Can't say I much like the idea of Component that only causes side-effects, but for sake of example, the implementation should match React-router more closely:

```cljs
(defn Redirect [props]
  (r/create-class
    {:component-did-mount (fn [this] (redirect! (r/props this)))
     :component-did-update (fn [this [_ prev-props]]
                             (if (not= (r/props this) prev-props)
                               (redirect! (r/props this))))
     :render (fn [this] nil)})

Reply:
Thanks for the review!

I did change the example to use this implementation. However I'm curious about in what situation does `:component-did-update` get called? I'd expect that `redirect!` on mount would take the user elsewhere and make update scenario redundant?